web/src/main/java/org/springframework/security/web/util/matcher/IpAddressMatcher.java

```diff
@@ -76,6 +80,8 @@ public boolean matches(String address) {
 		byte[] reqAddr = requiredAddress.getAddress();
 
 		int nMaskFullBytes = nMaskBits / 8;
+		Assert.isTrue(remAddr.length >= nMaskFullBytes,
```

Comment:
Actually, this check can be removed, correct @clementkng? You and I could not find a use case where this check would fail.

I'd recommend taking it out; however, if it stays, then let's make sure that the error messages align to both include the ip address and the bitmask length.

Reply:
@jzheaux good catch, forgot about that! I removed the check.